### DISCUSSION TITLE
Combine both cron and HTTP triggers in the same deta micro don't work

### DISCUSSION THREAD

**Discussion by egave** [deta/cloud-docs#288] [Help] [answered]
Hi,
I'm trying to implement the sample given in deta documentation [https://docs.deta.sh/docs/micros/cron](url) to combine both cron and HTTP triggers in the same deta micro.
But it doesn't work as the use of deta library don't give us the ability to use **app.get** or **app.post** for the HTTP trigger.
'App' Deta interface only seem to have **lib.cron** or **lib.run** definition.
Any idea what I am doing wrong ?

```
import { App } from 'deta'
import express from 'express'
import { webhookCallback } from 'grammy'
import { bot } from './bot'

const app = App(express());
//const app = express();

app.use(express.json())

app.post('/' + process.env.SECRET_PATH, webhookCallback(bot, 'express'))

app.lib.cron((event: any) => {
    return 'Hello Deta, I am a cron job';
});

module.exports = app;
```
Best regards,
Enguerrand

**Comment by abdelhai** [comment]
when you create a micro, you get an http endpoint which you would use to invoke your http code. you will find it again in your micro's details page or by typing `deta details` in your terminal (in the root of your micro)

**Reply by egave** [reply]
Thanks for your reply @abdelhai . But I succeeded to create an endpoint when creating a deta micro-service. What I would like to achieve is to combine the use of a CRON and endpoint together in a same Micro-Service as presented in [https://docs.deta.sh/docs/micros/cron](https://github.com/orgs/deta/discussions/url). The sample code provided in the docs don't work for me.
I have the following errors :

> Property 'use' does not exist on type 'DetaLib'.
> Property 'post' does not exist on type 'DetaLib'.

Any idea ?

**Reply by aavshr** [reply]
hi @egave , seems like using `import` might be the issue here. Unfortunately Micros do not support es modules yet. Can you please use `require` and try again? 

```nodejs
const { App } = require('deta');
const express = require('express');

const app = App(express());
```

**Comment by egave** [comment]
Hi @aavshr ! Thank you for your help.  I just give it a try on your solution. But I have the following error on Visor logs :
`TypeError: express is not a function
    at Object.<anonymous> (/var/task/dist/index.js:8:17)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:101:18)`

Here is my code, once transcripted it in javascript :
```"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const { App } = require('deta');
const { express } = require('express');
const grammy_1 = require("grammy");
const bot_1 = require("./bot");

const app = App(express());

app.use(express.json());
app.post('/' + process.env.SECRET_PATH, (0, grammy_1.webhookCallback)(bot_1.bot, 'express'));
app.lib.cron((event) => {
    return 'Hello Deta, I am a cron job';
});
module.exports = app;
```
Anyway, I finally have opted for a second micro-service to implement the CRON :-))

**Reply by aavshr** [reply]
hello @egave , this was because of  the destructured import for express `const {express } = require('express')` instead of `const express = require('express')`. Hopefully you had no issues with the second `Micro` for the cron.

**Comment by abdelhai** [comment, marked as answer]
the answer: https://github.com/deta/docs/discussions/288#discussioncomment-3048878
